In [1]:
from __future__ import division
import numpy as np
import os, sys
import matplotlib
#matplotlib.use('Agg')
%matplotlib nbagg
import matplotlib.pyplot as plt
import pandas as pd
from collections import defaultdict

In [2]:
def getCDF(data):
    xdata = np.sort(data)
    ydata = [i/len(xdata) for i in range(len(xdata))]
    return xdata, ydata

In [3]:
DNS_lookups = {
"Nest Thermostat": ["transport.home.nest.com", "frontdoor.nest.com.", "time.nest.com.", "weather.nest.com.", "devices.nest.com."],
"SmartThings": ["pool.ntp.org.", "DC.connect.smartthings.com."],
"Pixstar Photoframe": ["api.pix-star.com.", "spheris.connectify.", "iptime.pix-star.com."],
"Nest Cam": ["nexus.dropcam.com.", "oculus*.dropcam.com.", "files.dropcam.com.", "pool.ntp.org."],
"Amazon Echo": ["amzdigitaldownloads.edgesuite.net.", "softwareupdates.amazon.com.", "todo-*.amazon.com.", "ntp.amazon.com.", "pindorama.amazon.com.",  "firs-*.amazon.com.", "ntp-*.amazon.com.", "*.spotify.com.", "guipitan.amazon.com.", "*.north-america.pool.ntp.org.", "spectrum.s3.amazonaws.com.", "pins.amazon.com.", "device-metrics-us.amazon.com."],
"Sharx IPCam": ["www.sharxsecurity.com.", "time.nist.gov."],
"Ubi": ["clients3.google.com.", "api.grooveshark.com.", "dl.google.com.", "www.google.ca.", "*.grooveshark.com.", "www.earthtools.org.", "play.googleapis.com.", "ip-api.com.", "android.clients.google.com.", "mtalk.google.com.", "portal.theubi.com.", "*.android.pool.ntp.org.", "www.google.com."]
}

# OTHER INTERESTING DNS: https://www.nanog.org/sites/default/files/Nadji.pdf

In [4]:
DNS_moderate_lookups = {
    'nest_therm': '.nest.com',
    'nest_cam': '.dropcam.com',
    'smartthings': '.smartthings.com',
    'pixstar': '.pix-star.com',
    'echo': 'device-metrics-us.amazon.com',
    'sharx': 'sharxsecurity.com',
    'august': '.august.com',
    'hue': '.meethue.com',
    #'apple': 'time-ios.apple.com',
    'apple': 'gs-loc.ls-apple.com',
    'ps': '.playstation.net',
    'xbox': 'xboxlive.com',
    'nintendo': 'nintendo.net',
}

DNS_strict_lookups = {
    'nest_therm': 'devices.nest.com',
    'nest_cam': 'nexus.dropcam.com',
    'smartthings': 'api.smartthings.com',
    'pixstar': 'api.pix-star.com',
    'echo': 'device-metrics-us.amazon.com',
    'sharx': 'sharxsecurity.com',
    'august': 'doorbells.august.com',
    'hue': 'bridge.meethue.com',
    'apple': 'gs-loc.ls-apple.com',
    'ps': 'update.playstation.net',
    'xbox': '.xboxlive.com',
    'nintendo': '.nintendo.net',
}

In [5]:
df = pd.read_csv("pima-extract.csv", sep="|").fillna('')

In [6]:
df.head()

frame.time_relative            eth.src            eth.dst         ip.src  \
0             0.000000  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c    24.9.76.250   
1             0.000001  00:1c:73:69:93:01  e0:ac:f1:39:39:53   69.241.95.82   
2             0.000005  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30    24.9.76.250   
3             0.000009  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0    24.9.76.250   
4             0.000010  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68  73.153.66.183   

          ip.dst                dns.qry.name  
0  69.252.212.78  android.clients.google.com  
1  69.31.113.189      e3645.b.akamaiedge.net  
2  69.252.212.88  android.clients.google.com  
3  69.252.212.84  android.clients.google.com  
4  69.252.212.76      www.polls.newsvine.com

# STATS

In [7]:
print "Total time of dns record [677 MB / Comcast-PIMA]:", (df.iloc[-1]['frame.time_relative'] - df.iloc[0]['frame.time_relative']), "s"
print "Total number of queries: ", len(df)
print "Queries without source IP (are probably IPv6 so IP was not extracted): ", len(df[df['ip.src']==''])

print "Unique source IPs (homes):", len(df['ip.src'].unique())
print "Unique destination IPs (DNS servers):", len(df['ip.dst'].unique())

print "Unique queries:", len(df['dns.qry.name'].unique())

Total time of dns record [677 MB / Comcast-PIMA]: 59.990207 s
Total number of queries:  2527293
Queries without source IP (are probably IPv6 so IP was not extracted):  502648
Unique source IPs (homes): 260841
Unique destination IPs (DNS servers): 11767
Unique queries: 129065


# To Do
- Why are there 11767 unique servers?
- Hypothesis: home-dns popularity based on threshold to filter out DNS
    - there will not be many homes with IoT related DNS [use second level domain (*.iot.com)]
    - homes with IoTs will have similar queries (individual queries not in other IoT homes can be removed)
    - each IoT device will have at least 1 regular (heartbeat) IoT query (not *.iot.com)
- Tree
- Related Work

# MAIN: Interesting Questions
- A way to blacklist popular non-IoT related domains
- Abnormal/malicious domain queries check
- Time based profile per device DNS query [ srcIP | domain | mean (T) | var (T) ]

## Time profile of DNS queries per IP

In [8]:
gp = df.groupby(['ip.src', 'dns.qry.name'])

In [9]:
qry_time_mean = gp['frame.time_relative'].mean()
qry_time_std = gp['frame.time_relative'].std()
qry_count = gp['frame.time_relative'].count()

In [10]:
qry_time_mean

ip.src         dns.qry.name                                                                                                                                            
                f5.sensr.net                                                                                                                                               29.860547
               $PROXY.utopia.net                                                                                                                                           33.304035
               %20http.hsd1.co.comcast.net                                                                                                                                 54.459351
               %24staticdomain                                                                                                                                             31.938133
               %24staticdomain.hsd1.co.comcast.net                                                          

## Time profile of known IoT based DNS queries per IP
- many IoTs (like Amazon Echo) seem to be using IPv6. Current trace only extracted IPv4 so limit analysis to valid source IPs for now

In [11]:
iot_list = defaultdict(int)
for domain in DNS_moderate_lookups.values():
    if len(iot_list) == 0:
        iot_list = df['dns.qry.name'].str.contains(domain)
    else:
        iot_list = iot_list | df['dns.qry.name'].str.contains(domain)
df_iot = df[iot_list]

In [12]:
df_iot.head()

frame.time_relative            eth.src            eth.dst         ip.src  \
156             0.425989  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c  173.14.26.110   
169             0.426302  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0  71.229.198.23   
290             0.428878  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c    69.244.12.6   
369             0.430517  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   24.9.133.179   
822             0.441398  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   67.177.225.0   

            ip.dst                      dns.qry.name  
156  69.252.212.80  fus01.ps4.update.playstation.net  
169  69.252.212.84  log-rts11-iad01.devices.nest.com  
290  69.252.212.78                   www.meethue.com  
369  69.252.212.74             www.sharxsecurity.com  
822  69.252.212.78            licensing.xboxlive.com

### Add corresponding device column based on query

In [13]:
inverse_DNS_lookups = {v: k for k, v in DNS_moderate_lookups.items()}
df_iot['device'] = ''

for domain in DNS_moderate_lookups.values():
    df_iot.loc[ df_iot['dns.qry.name'].str.contains(domain), 'device'] = inverse_DNS_lookups[domain]

/usr/local/lib/python2.7/site-packages/ipykernel/__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
/usr/local/lib/python2.7/site-packages/pandas/core/indexing.py:420: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


In [14]:
df_iot.head(10)

frame.time_relative            eth.src            eth.dst  \
156              0.425989  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
169              0.426302  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
290              0.428878  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
369              0.430517  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
822              0.441398  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
881              0.442581  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
882              0.442584  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
948              0.444548  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1058             0.446752  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1065             0.446888  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   

             ip.src         ip.dst  \
156   173.14.26.110  69.252.212.80   
169   71.229.198.23  69.252.212.84   
290     69.244.12.6  69.252.212.78   
369    24.9.133.179  69.252.212.74   
822    67.177.225.0  69.252.212.78   
881                                  
882                                  
948   174.51.70.150  69.252.212.88   
1058  71.229.163.73  69.252.212.88   
1065                                 

                                      dns.qry.name      device  
156               fus01.ps4.update.playstation.net          ps  
169               log-rts11-iad01.devices.nest.com  nest_therm  
290                                www.meethue.com         hue  
369                          www.sharxsecurity.com       sharx  
822                         licensing.xboxlive.com        xbox  
881                  sessiondirectory.xboxlive.com        xbox  
882                  sessiondirectory.xboxlive.com        xbox  
948                 gs-loc.ls-apple.com.akadns.net       apple  
1058  static-resource.np.community.playstation.net          ps  
1065                        xsts.auth.xboxlive.com        xbox

### Extract only IPv4-DNS

In [15]:
df_iot_ipv4 = df_iot[df_iot['ip.src']!='']
print "Known IoT queries: ", len(df_iot)
print "Known IoT queries with valid IPv4: ", len(df_iot_ipv4)

Known IoT queries:  18316
Known IoT queries with valid IPv4:  11863


In [16]:
print "List of unique IoT domains extracted (IPv4 only):",
df_iot_ipv4['dns.qry.name'].unique()

List of unique IoT domains extracted (IPv4 only):

array(['fus01.ps4.update.playstation.net',
       'log-rts11-iad01.devices.nest.com', 'www.meethue.com',
       'www.sharxsecurity.com', 'licensing.xboxlive.com',
       'gs-loc.ls-apple.com.akadns.net',
       'static-resource.np.community.playstation.net',
       'data-vef.xboxlive.com', 'data-vef.gtm.xboxlive.com',
       'us-prof.np.community.playstation.net', 'xsts.auth.xboxlive.com',
       'sessiondirectory.xboxlive.com', 'asm.np.community.playstation.net',
       'device-metrics-us.amazon.com', 'nsx.np.dl.playstation.net',
       'log-rts02-iad01.devices.nest.com',
       'gs2.ww.prod.dl.playstation.net', 'title.mgt.xboxlive.com',
       'image.xboxlive.com.nsatc.net', 'social.gtm.xboxlive.com',
       'avatarservices.gtm.xboxlive.com',
       'ps4-system.sec.np.dl.playstation.net',
       'transport01-rts09-iad01.transport.home.nest.com',
       'event.api.np.km.playstation.net', 'settings.xboxlive.com',
       'olvus.cdn.nintendo.net', 'msg.xboxlive.com',
       'msg.gtm.xbox

In [18]:
# SAVE IOT DNS DATAFRAME
df_iot_ipv4.to_pickle('processed/df_iot_device_ipv4.pkl')

In [167]:
gp = df_iot_ipv4.groupby(['ip.src', 'dns.qry.name', 'device'])

In [178]:
iot_count = gp['frame.time_relative'].count()

In [191]:
iot_count.sort(ascending=False)
iot_count.head(10)

/usr/local/lib/python2.7/site-packages/ipykernel/__main__.py:1: FutureWarning: sort is deprecated, use sort_values(inplace=True) for for INPLACE sorting
  if __name__ == '__main__':


ip.src         dns.qry.name                         device     
69.244.4.108   graph.api.smartthings.com            smartthings    123
75.70.226.15   gameclipscontent-d3008.xboxlive.com  xbox            27
               gameclipscontent-d2013.xboxlive.com  xbox            14
73.24.144.121  xsts.auth.xboxlive.com               xbox            13
71.205.176.37  cs.prd.msys.playstation.net          ps              11
73.95.103.156  device.auth.xboxlive.com             xbox            11
75.70.226.15   avatarservices.xboxlive.com          xbox            11
70.90.112.54   log-rts18-iad01.devices.nest.com     nest_therm      10
73.24.144.121  profile.xboxlive.com                 xbox            10
73.95.103.156  title.mgt.xboxlive.com               xbox             9
Name: frame.time_relative, dtype: int64

In [180]:
iot_mean_t = gp['frame.time_relative'].mean()
iot_std_t = gp['frame.time_relative'].std()

In [188]:
iot_mean_t.head()

ip.src        dns.qry.name                    device    
107.0.87.50   home.nest.com                   nest_therm     8.279527
              oculus877-vir.dropcam.com       nest_cam      38.292396
107.1.83.162  gs-loc.ls-apple.com.akadns.net  apple         57.600348
107.2.0.212   assets1.xboxlive.com            xbox          13.591052
              gs2.ww.prod.dl.playstation.net  ps            11.417153
Name: frame.time_relative, dtype: float64

In [189]:
iot_std_t.head()

ip.src        dns.qry.name                    device    
107.0.87.50   home.nest.com                   nest_therm         NaN
              oculus877-vir.dropcam.com       nest_cam           NaN
107.1.83.162  gs-loc.ls-apple.com.akadns.net  apple              NaN
107.2.0.212   assets1.xboxlive.com            xbox          0.000003
              gs2.ww.prod.dl.playstation.net  ps                 NaN
Name: frame.time_relative, dtype: float64

### Combine IoT dataframe as [ip, dns, device, count, mean, std]

In [196]:
iot_summary = pd.concat({'count': iot_count, 'mean': iot_mean_t, 'std': iot_std_t}, axis=1).reset_index()

#### Device vs number of queries
- Overall most queries: xbox, ps, echo, apple, nest_therm, nintendo, smartthings, nest_cam, hue, august, pixstar, sharx
- average queries per IP for each device

In [204]:
iot_summary.groupby('device')['count'].sum()

device
apple          1342
august           30
echo           2286
hue              39
nest_cam         82
nest_therm      999
nintendo        608
pixstar          22
ps             2981
sharx             7
smartthings     190
xbox           3277
Name: count, dtype: int64

#### IP vs Number of Queries

In [212]:
g1 = iot_summary.groupby(['ip.src', 'device'])['count'].sum().reset_index()
g2 = g1.pivot(index='ip.src', columns='device', values='count').fillna(0)
g2

device         apple  august  echo  hue  nest_cam  nest_therm  nintendo  \
ip.src                                                                    
107.0.87.50        0       0     0    0         1           1         0   
107.1.83.162       1       0     0    0         0           0         0   
107.2.0.212        0       0     0    0         0           0         0   
107.2.1.145        2       0     0    0         0           0         0   
107.2.1.64         0       0     0    0         0           0         0   
107.2.1.73         2       0     0    0         0           0         0   
107.2.10.205       1       0     0    0         0           0         0   
107.2.11.123       0       0     0    0         0           0         0   
107.2.12.105       0       0     1    0         0           0         0   
107.2.12.8         0       0     1    0         0           0         0   
107.2.128.43       1       0     0    0         0           0         0   
107.2.13.162       0       0     1    0         0           0         0   
107.2.13.182       0       0     0    0         0           0         0   
107.2.131.149      0       0     1    0         0           0         0   
107.2.134.125      0       0     0    0         0           0         0   
107.2.135.104      1       0     1    0         0           0         0   
107.2.135.240      0       0     0    0         0           0         0   
107.2.135.86       0       0     0    0         0           0         0   
107.2.135.94       0       0     0    0         0           0         0   
107.2.136.151      0       0     1    0         0           0         0   
107.2.136.43       0       0     0    0         0           0         0   
107.2.137.88       0       0     0    0         0           0         0   
107.2.138.90       0       0     0    0         0           4         0   
107.2.139.40       0       0     0    0         0           1         0   
107.2.14.32        0       0     1    0         0           0         0   
107.2.140.184      0       0     1    0         0           0         0   
107.2.141.33       0       0     0    0         0           0         0   
107.2.143.118      0       0     0    0         0           0         1   
107.2.143.217      0       0     0    0         0           0         0   
107.2.143.42       0       0     0    0         0           0         0   
...              ...     ...   ...  ...       ...         ...       ...   
98.245.72.10       1       0     0    0         0           0         0   
98.245.72.122      0       0     1    0         0           0         0   
98.245.74.177      0       0     1    0         0           0         0   
98.245.74.215      0       0     0    0         0           0         1   
98.245.74.241      1       0     0    0         0           0         0   
98.245.77.113      2       0     0    0         0           0         0   
98.245.78.15       1       0     0    0         0           0         0   
98.245.79.59       0       0     1    0         0           0         0   
98.245.8.238       0       0     0    0         0           0         0   
98.245.80.203      0       0     0    0         0           0         0   
98.245.82.238      0       0     0    0         0           0         1   
98.245.82.61       0       0     0    0         0           0         0   
98.245.83.226      0       0     0    0         0           0         0   
98.245.83.86       0       0     0    0         0           0         0   
98.245.84.125      0       0     0    0         0           0         0   
98.245.85.64       0       0     0    0         0           0         0   
98.245.87.30       0       0     0    0         0           0         1   
98.245.88.144      0       0     0    0         0           1         0   
98.245.89.101      1       0     0    0         0           0         0   
98.245.89.162      0       0     1    0         0           0         0   
98.245.9.201       0     

#### Average number of queries per IP for each device (biased)

In [221]:
g2.mean()

device
apple          0.208159
august         0.004653
echo           0.354584
hue            0.006049
nest_cam       0.012719
nest_therm     0.154956
nintendo       0.094307
pixstar        0.003412
ps             0.462386
sharx          0.001086
smartthings    0.029471
xbox           0.508298
dtype: float64

#### Average number of queries per IP for each device for only those IPs with the device (unbiased)
- most queries from devices: sharx, smartthings, xbox, ps, august, nest_therm, nintendo, nest_cam, pixstar, hue, apple, echo

In [220]:
g1.groupby('device')['count'].mean()

device
apple          1.223336
august         2.307692
echo           1.118395
hue            1.300000
nest_cam       1.782609
nest_therm     2.134615
nintendo       1.993443
pixstar        1.692308
ps             2.409863
sharx          3.500000
smartthings    3.015873
xbox           2.471342
Name: count, dtype: float64

### Remove IPs with query count = 1
- mean, std of time only makes sense for repeated queries

In [199]:
iot_many_query = iot_summary[iot_summary['count']>1]

print "Total IoT related queries: ", len(df_iot_ipv4)
print "Unique IoT related IP-DNS combinations: ", len(iot_summary)
print "Unique IoT related IP-DNS combinations with only 1 query: ", len(iot_summary[iot_summary['count']==1])
print "Unique IoT related IP-DNS combinations with more than 1 query: ", len(iot_many_query)

Total IoT related queries:  11863
Unique IoT related IP-DNS combinations:  9736
Unique IoT related IP-DNS combinations with only 1 query:  8246
Unique IoT related IP-DNS combinations with more than 1 query:  1490


In [222]:
iot_many_query

ip.src                                     dns.qry.name  \
3        107.2.0.212                             assets1.xboxlive.com   
5        107.2.1.145                   gs-loc.ls-apple.com.akadns.net   
7         107.2.1.73                   gs-loc.ls-apple.com.akadns.net   
14      107.2.13.182                           licensing.xboxlive.com   
31      107.2.135.94                             xflight.xboxlive.com   
36      107.2.138.90      transport01.rts18.iad01.production.nest.com   
44      107.2.144.55                        bravia.dl.playstation.net   
48     107.2.150.198                   gs-loc.ls-apple.com.akadns.net   
49     107.2.150.222                          update-cdn.xboxlive.com   
50     107.2.151.241                             doorbells.august.com   
53     107.2.152.116                  gs-sec.ww.np.dl.playstation.net   
56     107.2.152.221                           xsts.auth.xboxlive.com   
61     107.2.157.126                   gs-loc.ls-apple.com.akadns.net   
67     107.2.159.222                           licensing.xboxlive.com   
69      107.2.16.174                     device-metrics-us.amazon.com   
70       107.2.16.33                          update-cdn.xboxlive.com   
71       107.2.16.33                              update.xboxlive.com   
72       107.2.16.33                           xsts.auth.xboxlive.com   
88      107.2.17.247                      cs.prd.msys.playstation.net   
96      107.2.174.39                   gs-loc.ls-apple.com.akadns.net   
97     107.2.175.139                     device-metrics-us.amazon.com   
111     107.2.183.97                           xsts.auth.xboxlive.com   
116     107.2.19.244                   gs-loc.ls-apple.com.akadns.net   
131     107.2.198.58             ps4-system.sec.np.dl.playstation.net   
135    107.2.199.124                             assets1.xboxlive.com   
139     107.2.20.143                    sessiondirectory.xboxlive.com   
162     107.2.214.30                        nsx.np.dl.playstation.net   
166    107.2.218.179               commerce.api.np.km.playstation.net   
167    107.2.218.179                  gs-sec.ww.np.dl.playstation.net   
192    107.2.229.117               commerce.api.np.km.playstation.net   
...              ...                                              ...   
9435    98.225.69.43                             assets1.xboxlive.com   
9450    98.225.72.95                 log-rts12-iad01.devices.nest.com   
9451    98.225.72.95  transport01-rts12-iad01.transport.home.nest.com   
9452    98.225.72.95                                 weather.nest.com   
9454   98.225.73.136                            settings.xboxlive.com   
9455   98.225.73.227                     device-metrics-us.amazon.com   
9456    98.225.73.72                             a.ad.playstation.net   
9457    98.225.73.72                        nsx.np.dl.playstation.net   
9461   98.225.75.240                   gs-loc.ls-apple.com.akadns.net   
9476   98.225.79.226                            settings.xboxlive.com   
9486   98.225.85.113                           licensing.xboxlive.com   
9498    98.225.89.19                           user.auth.xboxlive.com   
9506   98.225.92.110             us-prof.np.community.playstation.net   
9508   98.225.93.215                 asm.np.community.playstation.net   
9509   98.225.93.215              us-ivt.np.community.playstation.net   
9512    98.225.95.10                     device-metrics-us.amazon.com   
9513   98.225.95.162                   gs-loc.ls-apple.com.akadns.net   
9523   98.245.110.12                         device.auth.xboxlive.com   
9526  98.245.114.123                             a.ad.playstation.net   
9532  98.245.126.179                        nsx.np.dl.playstation.net   
9536  98.245.128.178                     device-metrics-us.amazon.com   
9541   98.245.132.10                   gs-loc.ls-apple.com.akadns.net   
9574  98.245.169.198                     device-metrics-us.amazon.com   

In [195]:
iot_summary

count  \
ip.src        dns.qry.name                                       device              
107.0.87.50   home.nest.com                                      nest_therm      1   
              oculus877-vir.dropcam.com                          nest_cam        1   
107.1.83.162  gs-loc.ls-apple.com.akadns.net                     apple           1   
107.2.0.212   assets1.xboxlive.com                               xbox            2   
              gs2.ww.prod.dl.playstation.net                     ps              1   
107.2.1.145   gs-loc.ls-apple.com.akadns.net                     apple           2   
107.2.1.64    update-cdn.xboxlive.com                            xbox            1   
107.2.1.73    gs-loc.ls-apple.com.akadns.net                     apple           2   
107.2.10.205  gs-loc.ls-apple.com.akadns.net                     apple           1   
107.2.11.123  bravia-e.dl.playstation.net                        ps              1   
107.2.12.105  device-metrics-us.amazon.com                       echo            1   
107.2.12.8    device-metrics-us.amazon.com                       echo            1   
107.2.128.43  gs-loc.ls-apple.com.akadns.net                     apple           1   
107.2.13.162  device-metrics-us.amazon.com                       echo            1   
107.2.13.182  licensing.xboxlive.com                             xbox            2   
107.2.131.149 device-metrics-us.amazon.com                       echo            1   
107.2.134.125 eplists.xboxlive.com                               xbox            1   
              www.xboxlive.com                                   xbox            1   
107.2.135.104 device-metrics-us.amazon.com                       echo            1   
              gs-loc.ls-apple.com.akadns.net                     apple           1   
107.2.135.240 attestation.xboxlive.com                           xbox            1   
              title.mgt.xboxlive.com                             xbox            1   
              update-cdn.xboxlive.com                            xbox            1   
              update.xboxlive.com                                xbox            1   
              xsts.auth.xboxlive.com                             xbox            1   
107.2.135.86  52-40-147-158-push.np.communication.playstation... ps              1   
              asm.np.community.playstation.net                   ps              1   
              auth.api.np.ac.playstation.net                     ps              1   
              fus01.ps4.update.playstation.net                   ps              1   
              us-push.np.communication.playstation.net           ps              1   
...                                                                            ...   
98.245.74.177 device-metrics-us.amazon.com                       echo            1   
98.245.74.215 npdi.cdn.nintendo.net                              nintendo        1   
98.245.74.241 gs-loc.ls-apple.com.akadns.net                     apple           1   
98.245.77.113 gs-loc.ls-apple.com.akadns.net                     apple           2   
98.245.78.15  gs-loc.ls-apple.com.akadns.net                     apple           1   
98.245.79.59  device-metrics-us.amazon.com                       echo            1   
98.245.8.238  gs2.ww.prod.dl.playstation.net                     ps              1   
98.245.80.203 update-cdn.xboxlive.com                            xbox            1   
98.245.82.238 nppl.app.nintendo.net                              nintendo        1   
98.245.82.61  us-prof.np.community.playstation.net               ps              1   
98.245.83.226 gs2.ww.prod.dl.playstation.net                     ps              1   
98.245.83.86  ps4-system.sec.np.dl.playstation.net               ps              1   
98.245.84.125 us-prof.np.community.playstation.net               ps              1   
98.245.85.64  us-prof.np.community.playstation.net               ps              1   
98.245.87.30  nppl.app.nintendo.net

# Extract known IoT-DNS by src IP

In [25]:
df_IP = pd.DataFrame({'ip.src': df['ip.src'].unique()}).sort_values(by='ip.src').set_index('ip.src')

In [27]:
ip_dns = defaultdict(int)
for domain in DNS_moderate_lookups.values():
    #ip_dns[domain] = df['dns.qry.name'].str.contains(domain)
    temp = pd.DataFrame( { 'ip.src' : df[ df['dns.qry.name'].str.contains(domain) ]['ip.src'].unique(),
              domain: True} ).set_index('ip.src')
    df_IP[domain]=temp
    df_IP[domain].fillna(False, inplace=True)

In [28]:
df_IP.head()

.meethue.com gs-loc.ls-apple.com device-metrics-us.amazon.com  \
ip.src                                                                        
                      True                True                         True   
106.184.1.178        False               False                        False   
107.0.141.78         False               False                        False   
107.0.143.131        False               False                        False   
107.0.164.246        False               False                        False   

              sharxsecurity.com nintendo.net .dropcam.com .august.com  \
ip.src                                                                  
                          False         True         True        True   
106.184.1.178             False        False        False       False   
107.0.141.78              False        False        False       False   
107.0.143.131             False        False        False       False   
107.0.164.246             False        False        False       False   

              .playstation.net .smartthings.com .pix-star.com xboxlive.com  \
ip.src                                                                       
                          True             True         False         True   
106.184.1.178            False            False         False        False   
107.0.141.78             False            False         False        False   
107.0.143.131            False            False         False        False   
107.0.164.246            False            False         False        False   

              .nest.com  
ip.src                   
                   True  
106.184.1.178     False  
107.0.141.78      False  
107.0.143.131     False  
107.0.164.246     False

### CHECK particular IP

In [29]:
df_IP.ix['98.225.66.206']

.meethue.com                    False
gs-loc.ls-apple.com             False
device-metrics-us.amazon.com    False
sharxsecurity.com               False
nintendo.net                    False
.dropcam.com                    False
.august.com                     False
.playstation.net                False
.smartthings.com                False
.pix-star.com                   False
xboxlive.com                    False
.nest.com                       False
Name: 98.225.66.206, dtype: bool

### ANALYSIS

In [30]:
df_IP.sum()

.meethue.com                      31
gs-loc.ls-apple.com             1098
device-metrics-us.amazon.com    2045
sharxsecurity.com                  2
nintendo.net                     306
.dropcam.com                      47
.august.com                       14
.playstation.net                1238
.smartthings.com                  64
.pix-star.com                     13
xboxlive.com                    1327
.nest.com                        469
dtype: int64

#### sort source IPs by total number of DNS-IoT found in 1 min trace

Actual PIMA src IPs with IoTs = 17577 (current trace union another trace) from around 250k

In [34]:
indx = df_IP.sum(axis=1).sort_values(ascending=False).index

In [35]:
df_IP_sorted = df_IP.reindex(indx)
df_IP_sorted['total'] = df_IP_sorted.sum(axis=1)

In [40]:
print "Source IPs with IoTs:", len(df_IP_sorted[ df_IP_sorted.total>0 ])
print "Source IPs without IoTs:", len(df_IP_sorted[ df_IP_sorted.total==0 ])
print "Percentage of IoT source IPs in current trace: ", (len(df_IP_sorted[ df_IP_sorted.total>0 ])/len(df_IP_sorted[ df_IP_sorted.total==0 ]) * 100)

Source IPs with IoTs: 6448
Source IPs without IoTs: 254393
Percentage of IoT source IPs in current trace:  2.534660938


In [43]:
# Distribution of number of IoTs per home
tot = df_IP_sorted[ df_IP_sorted.total>0 ]['total']
tot.head(10)

ip.src
69.241.95.83     11
                 10
69.241.95.85     10
69.241.95.82     10
69.241.95.87      9
69.241.95.86      9
69.241.95.84      9
73.95.130.62      5
73.95.130.127     5
73.95.130.28      5
Name: total, dtype: int64

In [44]:
df_IP_sorted.to_pickle("pima-extract-sorted-IoT-IPs.pkl")

In [45]:
from collections import Counter
print "Distribution of number of IoTs per home: ", Counter(tot)

Distribution of number of IoTs per home:  Counter({1: 6345, 2: 72, 4: 11, 5: 7, 3: 6, 9: 3, 10: 3, 11: 1})


In [46]:
interesting = df_IP_sorted[df_IP_sorted.total > 2]
print len(interesting)
interesting.sum().sort_values(ascending=False)

31


total                           165
xboxlive.com                     31
.playstation.net                 30
device-metrics-us.amazon.com     29
gs-loc.ls-apple.com              24
nintendo.net                     15
.nest.com                        10
.august.com                       7
.dropcam.com                      7
.smartthings.com                  6
.pix-star.com                     4
.meethue.com                      2
sharxsecurity.com                 0
dtype: float64

- Most popular Xboxlive, Playstation, Amazon Echo
- Followed by Nintendo, Apple Watch, Nest thermostat
- August Lock, Dropcam, Smartthings


### SUMMARY 2: USING STRICT DOMAINS instead of MODERATE

In [47]:
df_IP2 = pd.DataFrame({'ip.src': df['ip.src'].unique()}).sort_values(by='ip.src').set_index('ip.src')

ip_dns2 = defaultdict(int)
for domain in DNS_strict_lookups.values():
    #ip_dns[domain] = df['dns.qry.name'].str.contains(domain)
    temp = pd.DataFrame( { 'ip.src' : df[ df['dns.qry.name'].str.contains(domain) ]['ip.src'].unique(),
              domain: True} ).set_index('ip.src')
    df_IP2[domain]=temp
    df_IP2[domain].fillna(False, inplace=True)

In [48]:
df_IP2.sum()

bridge.meethue.com                15
gs-loc.ls-apple.com             1098
device-metrics-us.amazon.com    2045
sharxsecurity.com                  2
.nintendo.net                    306
nexus.dropcam.com                 14
doorbells.august.com              13
update.playstation.net           197
api.smartthings.com                6
api.pix-star.com                  13
.xboxlive.com                   1327
devices.nest.com                 297
dtype: int64

### CHECK particular IP

In [49]:
df_IP2.ix['69.244.36.23']

bridge.meethue.com              False
gs-loc.ls-apple.com             False
device-metrics-us.amazon.com    False
sharxsecurity.com               False
.nintendo.net                   False
nexus.dropcam.com               False
doorbells.august.com            False
update.playstation.net          False
api.smartthings.com             False
api.pix-star.com                False
.xboxlive.com                   False
devices.nest.com                False
Name: 69.244.36.23, dtype: bool

In [50]:
#pd.concat( [ df['dns.qry.name'].str.contains(domain), df['ip.src'] ], axis=1 ).groupby('ip.src').any()

In [51]:
#df_ip_dns = pd.DataFrame(ip_dns).join(df['ip.src'])

In [52]:
#ip_dns = defaultdict(int)
#for domain in DNS_lookups.values():
#    ip_dns[domain] = df[ df['dns.qry.name'].str.contains(domain) ]['ip.src'].unique()

In [53]:
#ip_dns

In [54]:
#cols = list(df_ip_dns.columns)

In [55]:
#df_ip_dns

### NEST THERMOSTAT

In [56]:
nest_df = df[df['dns.qry.name'].str.contains("nest.com")]

nest_df

frame.time_relative            eth.src            eth.dst  \
169                0.426302  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
3089               0.495322  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
3372               0.501875  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
5564               0.555964  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
5586               0.556278  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
9870               0.656782  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
17405               0.83489  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
20185              0.897977  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
20192              0.898152  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
22159              0.943734  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
22160              0.943751  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
22178              0.943952  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
22180              0.943961  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
27467               1.06759  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
31919               1.17162  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
35068               1.24943  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
35084                1.2497  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
39620               1.35669  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
39855               1.36242  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
41170               1.39359  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
41172                1.3936  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
44676               1.48026  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
46262               1.52012  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
47320               1.54617  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
51673               1.65402  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
52456               1.67284  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
52463               1.67304  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
56561               1.77217  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
56664                1.7747  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
56830               1.77882  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
...                     ...                ...                ...   
2473621              58.746  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2476849             58.8193  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2487189             59.0522  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2487201             59.0525  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2498245             59.3113  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2498655             59.3208  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2498670              59.321  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2498848             59.3257  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2499703              59.345  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2499712             59.3452  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2500155             59.3557  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2501977             59.3948  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2503837             59.4379  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2503853             59.4381  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2504830             59.4594  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2504832             59.4596  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2506011             59.4891  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2507658             59.5277  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2507659             59.5277  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2507887             59.5335  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2509220             59.5645  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2510222             59.5863  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2510235             59.5866  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2510396             59.5896  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2510403             59.5898  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2513059             59.6505  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
251307

## NEST CAM

In [57]:
nestcam_df = df[df['dns.qry.name'].str.contains("dropcam.com")]

nestcam_df

frame.time_relative            eth.src            eth.dst  \
63693               1.94753  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
101781              2.86947  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
140529              3.79976  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
182371              4.77707  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
182814              4.78651  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
216806              5.59157  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
216814              5.59175  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
253130              6.43821  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
253135              6.43844  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
314078              7.89541  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
314081              7.89558  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
322556              8.09536  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
322786              8.10017  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
322794              8.10042  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
345830              8.64437  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
345841              8.64456  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
369327              9.20798  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
369349              9.20828  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
382607              9.52896  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
418582              10.3803  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
418589              10.3805  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
419607              10.4044  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
419608              10.4044  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
419613              10.4045  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
419618              10.4047  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
479977                11.82  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
479985              11.8202  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
536943              13.1968  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
536947               13.197  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
551971              13.5492  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
...                     ...                ...                ...   
1613907             38.7842  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1794653             42.8983  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
1822363             43.5593  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
1822374             43.5594  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1832520             43.7985  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1836509             43.8891  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1836516             43.8893  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1932528             46.1721  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1932541             46.1723  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1990751             47.5924  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1990759             47.5926  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2001089             47.8413  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2001101             47.8415  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2072297             49.4954  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2072308             49.4956  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2073369             49.5222  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2306997        54.943900000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2307023        54.944164000  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2331381        55.492544000  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2331387        55.492721000  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2350725        55.931941000  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2398776             56.9975  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2447525             58.1257  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2496181              59.263  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2496192             59.2633  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2511549             59.6156  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
251155

## SMARTTHINGS

In [58]:
smartthings_df = df[df['dns.qry.name'].str.contains("smartthings.com")]

smartthings_df

frame.time_relative            eth.src            eth.dst  \
35717               1.26441  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
36055               1.27235  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
77250               2.26884  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
77587               2.27695  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
97418               2.76591  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
97431               2.76615  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
118686              3.27225  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
119201              3.28495  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
161318              4.27683  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
162910              4.31061  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
197671              5.14072  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
203494              5.28028  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
203874              5.28834  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
224879              5.78291  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
246626              6.28504  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
246962              6.29292  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
253771              6.45258  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
253777              6.45273  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
259723              6.59727  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
289187              7.29928  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
289469              7.30637  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
331183              8.29289  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
331535              8.30087  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
332149                8.315  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
344034               8.6023  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
373064               9.2964  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
373065               9.2964  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
373413              9.30432  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
377468              9.40425  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
415435              10.3043  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
...                     ...                ...                ...   
2114488             50.4693  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2141568             51.0941  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2157629              51.467  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2157632              51.467  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2157939             51.4747  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2200617             52.4712  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2200941             52.4791  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2235109        53.273091000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2243578        53.472773000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2244051        53.482505000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2250667        53.635675000  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2269333        54.061787000  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2280231        54.318581000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2283587        54.393372000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2287089        54.477374000  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2287431        54.485309000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2330971        55.482748000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2331290        55.490883000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2359547             56.1288  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2376033             56.4872  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2376373             56.4951  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2420342              57.491  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2420675             57.4988  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2461040             58.4462  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2463102             58.4952  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2463419             58.5033  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
247296

## PIX-STAR

In [59]:
pixstar_df = df[df['dns.qry.name'].str.contains("pix-star.com")]

pixstar_df

frame.time_relative            eth.src            eth.dst  \
95755               2.72294  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
256828                6.528  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
339901              8.50164  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
339911              8.50186  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
346779              8.66756  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
347916               8.6936  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
348634              8.71155  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
348649              8.71172  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
350370              8.75358  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
578200               14.197  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
582585              14.3005  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1081257             26.2144  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
1452178             34.9854  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1585131             38.0962  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
1585133             38.0962  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
1585138             38.0963  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1882097             44.9832  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1886334             45.0849  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2373430              56.427  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2373616             56.4308  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2373622              56.431  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2373626             56.4311  00:1c:73:69:93:01  e0:ac:f1:39:39:53   

                 ip.src          ip.dst                          dns.qry.name  
95755       75.70.222.3   69.252.212.74                      api.pix-star.com  
256828   71.205.156.203   69.252.212.80                      api.pix-star.com  
339901     24.11.158.28   69.252.212.66                      api.pix-star.com  
339911     69.241.95.82  84.233.158.131                      api.pix-star.com  
346779     24.11.158.28   69.252.212.70  api.pix-star.com.hsd1.az.comcast.net  
347916     24.11.158.28   69.252.212.78                    api.pix-star.com.#  
348634     24.11.158.28   69.252.212.88                api.pix-star.com.wlan0  
348649     69.241.95.87     199.7.91.13                api.pix-star.com.wlan0  
350370     24.11.158.28   69.252.212.66                      api.pix-star.com  
578200    71.237.68.136   69.252.212.74                      api.pix-star.com  
582585    71.237.68.136   69.252.212.88                      api.pix-star.com  
1081257  50.134.184.119   69.252.212.76                      api.pix-star.com  
1452178  67.161.139.137   69.252.212.82                      api.pix-star.com  
1585131   69.244.62.119   69.252.212.74                      api.pix-star.com  
1585133   69.244.62.119   69.252.212.72                      api.pix-star.com  
1585138    69.241.95.83  84.233.158.131                      api.pix-star.com  
1882097      24.8.53.91   69.252.212.86                      api.pix-star.com  
1886334      24.8.53.91   69.252.212.80                api.pix-star.com.wlan0  
2373430   76.18.107.172   69.252.212.74                      api.pix-star.com  
2373616   76.18.107.172   69.252.212.80                      api.pix-star.com  
2373622    69.241.95.85    185.16.44.67                      api.pix-star.com  
2373626    69.241.95.85  84.233.158.131                      api.pix-star.com

## AMAZON ECHO

In [60]:
echo_df = df[df['dns.qry.name'].str.contains("device-metrics-us.amazon.com")]

echo_df


frame.time_relative            eth.src            eth.dst  \
2211                0.47402  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2682               0.485964  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2793               0.488358  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
3020               0.493752  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
3490               0.504294  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
4479               0.529533  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
4864                0.53921  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
6334                0.57313  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
6629               0.579893  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
6740               0.582915  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
6923               0.587124  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
7253               0.593887  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
7815               0.606623  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
8880               0.632884  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
9459               0.647431  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
9519               0.648779  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
9822                 0.6557  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
10255              0.665785  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
10296              0.667023  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
10378              0.669148  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
10595              0.674044  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
10856              0.679904  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
11250              0.689311  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
11356              0.692368  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
11829               0.70389  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
11860              0.704245  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
12020              0.708007  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
12214              0.712144  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
12452              0.717988  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
12479               0.71851  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
...                     ...                ...                ...   
2513023             59.6498  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2513460             59.6604  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2513493             59.6608  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2513524             59.6618  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2513607             59.6634  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2513803             59.6673  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2513854             59.6688  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2514029             59.6727  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2514504             59.6856  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2514952             59.6959  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2515168             59.7011  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2515195              59.702  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2515430             59.7083  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2518539             59.7841  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2518698             59.7875  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2519025             59.7956  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2519265             59.8013  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2519874             59.8164  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2520685             59.8349  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2520950             59.8411  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2522932             59.8875  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2523669             59.9058  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2524008             59.9132  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2524987             59.9374  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2525022             59.9382  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2525205             59.9423  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
252665

## SHARX IPCAM

In [61]:
sharx_df = df[df['dns.qry.name'].str.contains("sharxsecurity.com")]

sharx_df

frame.time_relative            eth.src            eth.dst  \
369                0.430517  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
556091              13.6497  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
853647              20.7922  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
1218129             29.4658  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
1639121             39.3657  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
1826859             43.6617  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2130890             50.8498  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   

               ip.src         ip.dst           dns.qry.name  
369      24.9.133.179  69.252.212.74  www.sharxsecurity.com  
556091   24.9.133.179  69.252.212.82  www.sharxsecurity.com  
853647   24.9.133.179  69.252.212.70  www.sharxsecurity.com  
1218129   73.95.22.25  69.252.212.74  www.sharxsecurity.com  
1639121  24.9.133.179  69.252.212.78  www.sharxsecurity.com  
1826859  24.9.133.179  69.252.212.84  www.sharxsecurity.com  
2130890  24.9.133.179  69.252.212.70  www.sharxsecurity.com

##  UBI

In [62]:
ubi_df = df[df['dns.qry.name'].str.contains("theubi.com")]

ubi_df


Empty DataFrame
Columns: [frame.time_relative, eth.src, eth.dst, ip.src, ip.dst, dns.qry.name]
Index: []

## AUGUST LOCK

In [63]:
august_df = df[df['dns.qry.name'].str.contains("august.com")]

august_df


frame.time_relative            eth.src            eth.dst  \
56343               1.76712  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
56349               1.76737  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
77456               2.27393  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
236160              6.03875  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
301253               7.5919  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
301254               7.5919  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
333593              8.34986  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
333805              8.35463  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
335296              8.38933  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
383895              9.55984  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
384116              9.56512  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
384122              9.56538  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
386718              9.62558  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
526817              12.9461  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
526824              12.9464  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
685921              16.7756  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
685929              16.7758  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
842335              20.5192  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
842343              20.5195  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1116124             27.0506  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1281129             30.9411  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
1281139             30.9414  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1283410             30.9928  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1283421              30.993  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1284473             31.0163  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1364551             32.9023  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1904450             45.5108  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1904837             45.5199  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
1909915             45.6418  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2120912             50.6199  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2120919               50.62  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2362220             56.1848  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2482041             58.9361  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2482264             58.9409  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2483621             58.9715  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   

                 ip.src         ip.dst  \
56343     107.2.151.241  69.252.212.80   
56349      69.241.95.85  205.251.195.2   
77456     174.51.52.254  69.252.212.80   
236160                                   
301253    107.2.151.241  69.252.212.88   
301254    107.2.151.241  69.252.212.68   
333593    174.51.52.254  69.252.212.80   
333805    174.51.52.254  69.252.212.76   
335296    174.51.52.254  69.252.212.78   
383895    67.190.56.235  69.252.212.74   
384116    67.190.56.235  69.252.212.72   
384122     69.241.95.83  205.251.195.2   
386718    67.190.56.235  69.252.212.66   
526817                                   
526824     69.241.95.82    66.94.9.119   
685921    67.190.35.159  69.252.212.76   
685929     69.241.95.84  205.251.195.2   
842335    73.243.173.42  69.252.212.88   
842343     69.241.95.87  205.251.195.2   
1116124                                  
1281129                                  
1281139    69.241.95.85  205.251.195.2   
1283410                                  
1283421    69.241.95.86  205.251.195.2   
1284473    69.241.95.86  205.251.195.2   
1364551   107.2.151.241  69.252.212.84   
1904450   73.78.159.243  69.252.212.86   
1904837   73.78.159.243  69.252.212.72   
1909915   73.78.159.243  69.252.212.72   
2120912   73.243.173.42  69.252.212.72   
2120919    69.241.95.83  205.251.195.2   
2362220   67.190.56.235  69.252.212.78   
2482041  71.205.191.241  69.252.212.70   
2482264  71.205.191.241  69.252.212.74   
2483621  71.205.191.241  69.252.212.74   

          

## PHILLIPS HUE

In [64]:
hue_df = df[df['dns.qry.name'].str.contains("meethue.com")]

hue_df


frame.time_relative            eth.src            eth.dst  \
290                0.428878  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
96133               2.73291  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
96141               2.73304  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
243352              6.20694  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
399630              9.92882  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
402097              9.98414  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
454705              11.2076  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
475484              11.7104  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
544799              13.3828  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
700825              17.1368  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
761528                 18.6  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
762009              18.6122  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
812582               19.824  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
870844              21.1895  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
870850              21.1897  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
881577              21.4487  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
886788              21.5751  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
966480               23.483  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
971395              23.6049  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1162229             28.1513  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
1219038             29.4886  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
1219044             29.4887  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1219046             29.4889  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1312090             31.6708  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1434182             34.5661  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1470279             35.4042  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1524084             36.6807  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
1524202             36.6835  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1617659             38.8724  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1662426             39.9113  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1719425             41.1338  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1719426             41.1338  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1743614             41.6841  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1893564              45.252  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
1953468             46.6828  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
1968524              47.052  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2048901             48.9599  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2126157             50.7393  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2166854             51.6836  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2234450        53.258567000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2236207        53.297662000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2298515        54.744358000  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2372689             56.4098  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   

                 ip.src         ip.dst        dns.qry.name  
290         69.244.12.6  69.252.212.78     www.meethue.com  
96133     71.205.42.196  69.252.212.68  bridge.meethue.com  
96141                                   bridge.meethue.com  
243352    73.24.242.243  69.252.212.82  bridge.meethue.com  
399630   73.153.137.118  69.252.212.88  bridge.meethue.com  
402097       24.9.91.88  69.252.212.68     www.meethue.com  
454705    73.24.242.243  69.252.212.68  bridge.meethue.com  
475484    50.170.153.17  69.252.212.78     www.meethue.com  
544799   73.243.104.162  69.252.212.66  bridge.meethue.com  
700825                                     www.meethue.com  
761528     24.8.143.254  69.252.212.68  bridge.meethue.com  
762009     76.25.65.247  69.252.212.88     www.meethue.com  
812582    76.120.121.62  69.252.212.84  bridge.meethue.com  
870844   50.243.138.186  69.252.212.80  bridge.meethue.com  
870850                                  bridge.meethue.c

## APPLE WATCH

In [65]:
#apple_df = df[df['dns.qry.name'].str.contains("time-ios.g.aaplimg.com")]
apple_df = df[df['dns.qry.name'].str.contains("gs-loc.ls-apple.com")]

apple_df


frame.time_relative            eth.src            eth.dst  \
948                0.444548  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1232               0.450624  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2531               0.482263  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
5301               0.549262  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
9388               0.645707  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
9428               0.646887  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
10036              0.660586  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
10271              0.666278  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
11401              0.693539  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
12502              0.718924  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
17582               0.83912  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
18902              0.868639  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
20593              0.907198  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
21382              0.925713  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
22057               0.94125  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
22527              0.951126  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
22528              0.951134  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
22698              0.954396  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
22830              0.957201  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
23143              0.963666  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
26707               1.04899  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
27613               1.07075  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
30089                1.1268  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
30257               1.13102  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
32251               1.18012  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
32387               1.18328  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
32590               1.18809  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
37689               1.31016  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
37834                1.3138  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
37849               1.31441  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
...                     ...                ...                ...   
2452760             58.2512  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2452770             58.2513  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2452844             58.2531  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2456747             58.3449  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2456748             58.3449  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2464728             58.5336  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2471651             58.7004  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2473466             58.7426  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2478921             58.8654  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2481014             58.9116  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2484861             58.9985  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2486237             59.0312  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2486543             59.0381  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2488513             59.0821  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2491163             59.1439  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2491489             59.1519  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2491491             59.1519  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2500444             59.3612  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2500521             59.3629  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2500615             59.3652  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2503994             59.4418  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2505208             59.4693  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2509255             59.5654  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2509557             59.5721  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2512720             59.6431  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2513016             59.6497  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
251318

## PLAYSTATION

In [66]:
ps_df = df[df['dns.qry.name'].str.contains("update.playstation.net")]

ps_df



frame.time_relative            eth.src            eth.dst  \
156                0.425989  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
23582              0.973793  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
24491              0.994319  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
29469               1.11212  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
36481               1.28197  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
49581               1.60432  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
55896               1.75579  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
72514               2.15917  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
73569               2.18296  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
85579               2.47275  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
88048               2.53379  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
102607              2.88842  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
106310              2.97669  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
121868              3.34912  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
122988              3.37506  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
132765              3.61218  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
134867              3.66537  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
145505              3.91365  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
151748              4.05574  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
162656              4.30505  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
164474              4.34706  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
167186              4.41052  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
174505              4.58301  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
203272              5.27545  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
208957              5.41096  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
221687              5.70561  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
222845              5.73501  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
232385              5.95381  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
232745              5.96236  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
249613              6.35546  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
...                     ...                ...                ...   
2171936              51.804  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2192984              52.292  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2223968             53.0165  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2231306        53.187299000  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2242705        53.451413000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2243045        53.459321000  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2248617        53.588587000  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2261443        53.885396000  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2263190        53.925903000  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2266903        54.008082000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2304534        54.884904000  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2307667        54.959500000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2314622        55.117823000  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2333074        55.530046000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2333174        55.532237000  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2336269        55.600117000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2337405        55.624829000  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2350649        55.929767000  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2360585             56.1514  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2367606             56.2974  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2393490             56.8768  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2397721             56.9733  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2409396             57.2386  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2416180              57.393  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2418744             57.4522  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2432237             57.7694  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
243433

## XBOX

In [67]:
xbox_df = df[df['dns.qry.name'].str.contains(".xboxlive.com")]

xbox_df


frame.time_relative            eth.src            eth.dst  \
822                0.441398  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
881                0.442581  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
882                0.442584  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1065               0.446888  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
1066               0.446914  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
1580               0.458873  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
1586                0.45889  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
1781               0.463833  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2217               0.474166  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2318               0.477154  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
3571               0.506419  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
3936               0.515781  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
4023               0.518105  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
4033                0.51836  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
4652               0.534101  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
4659               0.534393  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
4851               0.539009  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
4852               0.539013  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
4865               0.539237  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
5242                0.54769  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
7696               0.604146  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
8681               0.628528  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
9149               0.639582  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
9772               0.654688  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
9778               0.654973  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
9858                0.65649  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
9860               0.656497  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
9866               0.656689  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
10197              0.664686  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
10771              0.677458  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
...                     ...                ...                ...   
2516561             59.7363  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2516997             59.7472  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2517086             59.7494  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2518348             59.7799  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2518565             59.7848  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2518690             59.7873  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2518743             59.7885  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2518802             59.7901  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2518936             59.7931  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2519462             59.8061  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2519849             59.8154  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2519998             59.8194  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2522707             59.8821  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2523246             59.8954  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2523255             59.8955  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2523562             59.9028  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2524842              59.934  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2525054             59.9391  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2525299             59.9453  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2525529             59.9502  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2525710             59.9542  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2526200             59.9655  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2526288              59.967  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2526327              59.968  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2526430             59.9698  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2526435               59.97  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
252646

## NINTENDO

In [68]:
nintendo_df = df[df['dns.qry.name'].str.contains(".nintendo.net")]

nintendo_df


frame.time_relative            eth.src            eth.dst  \
9074               0.637796  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
9990               0.659816  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
9991               0.659816  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
10591              0.673846  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
12566              0.720612  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
13296              0.737724  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
14461              0.766818  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
14473               0.76707  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
14681               0.77176  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
14690              0.771927  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
15341              0.787224  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
15401              0.788265  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
15707              0.795458  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
25279                1.0149  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
32339               1.18199  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
32504               1.18602  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
35351               1.25627  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
38553               1.33188  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
53941               1.70901  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
53951               1.70928  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
56821               1.77844  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
58388               1.81832  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
59516               1.84529  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
59520               1.84555  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
59971               1.85682  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
60855               1.87774  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
63013               1.93083  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
63973               1.95424  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
64944               1.97616  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
68791               2.07291  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
...                     ...                ...                ...   
2333444        55.538416000  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2350463        55.925690000  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2357148        56.073161000  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2361031             56.1599  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2362771             56.1959  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2374895             56.4603  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2379587             56.5675  00:1f:a0:11:b5:3a  40:a8:f0:34:1f:e0   
2383925             56.6628  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2396436             56.9439  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2416352             57.3976  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2416366             57.3978  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2424896             57.5994  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2424907             57.5996  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2425856             57.6203  00:1c:73:69:93:01  e0:ac:f1:39:39:53   
2426661             57.6373  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
2430509             57.7278  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2437280             57.8877  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2449820              58.178  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2466439             58.5749  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2476971             58.8224  00:1f:a0:11:b5:3a  40:a8:f0:34:20:2c   
2489194              59.098  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2493886             59.2083  00:1f:a0:11:b5:3a  40:a8:f0:34:48:5c   
2496398              59.268  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2502397             59.4044  00:1f:a0:11:b5:3a  40:a8:f0:34:19:30   
2505735             59.4822  00:1f:a0:11:b5:3a  40:a8:f0:33:fa:68   
2509874             59.5793  00:1f:a0:11:b5:3a  40:a8:f0:34:35:dc   
250987